credentials/oauth_credentials_provider.go

```diff
@@ -0,0 +1,32 @@
+package credentials
+
+import (
+	"net/http"
+
+	"golang.org/x/oauth2"
+)
+
+type OAuthCredentialsProvider interface {
+	SetHeaders(r *http.Request) error
```

Comment:
In case we add more methods to CredentialsProvider, this should probably inherit from that interface.
```suggestion
	CredentialsProvider
```